pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -112,8 +111,9 @@ type ruleCache struct {
 	addressSetByGroup map[string]v1beta1.GroupMemberPodSet
 
 	policySetLock sync.RWMutex
-	// policySet is a set to store NetworkPolicy UID strings.
-	policySet sets.String
+	// policySetByGroup is a set to store NetworkPolicy members.
+	// It is a mapping from a NetworkPolicy UID string to a Pod each.
+	policySetByGroup map[string]*v1beta1.PodReference
```

Comment:
`PodReference` has nothing to do with `NetworkPolicy`, please don't associate them just to store policy namespace and name.

I have added policy namespace and name to struct `rule`, you can retrieve them from it.

Reply:
> `PodReference` has nothing to do with `NetworkPolicy`, please don't associate them just to store policy namespace and name.
> 
> I have added policy namespace and name to struct `rule`, you can retrieve them from it.

It is true that `PodReference` has nothing to do with `NetworkPolicy`.But what if a `NetworkPolicy` does not have rules associated? We can define another struct to store namespace and name.

Reply:
For your previous question, the rules' lifecycle is same as their policy, it's guaranteed there will be at least one rule when there's a policy.

Reply:
> For your previous question, the rules' lifecycle is same as their policy, it's guaranteed there will be at least one rule when there's a policy.

Yes, I confirmed it after checking codes involved. Sorry for trouble caused.